hudi-client/src/test/java/org/apache/hudi/common/HoodieTestDataGenerator.java

```diff
@@ -114,20 +114,20 @@ public static void writePartitionMetadata(FileSystem fs, String[] partitionPaths
    * Generates a new avro record of the above schema format, retaining the key if optionally provided.
    */
   public static TestRawTripPayload generateRandomValue(HoodieKey key, String commitTime) throws IOException {
-    GenericRecord rec = generateGenericRecord(key.getRecordKey(), "rider-" + commitTime, "driver-" + commitTime, 0.0);
+    GenericRecord rec = generateGenericRecord(key.getRecordKey(), "rider-" + commitTime, "driver-" + commitTime, 0);
     return new TestRawTripPayload(rec.toString(), key.getRecordKey(), key.getPartitionPath(), TRIP_EXAMPLE_SCHEMA);
   }
 
   /**
    * Generates a new avro record of the above schema format, retaining the key if optionally provided.
    */
   public static HoodieAvroPayload generateAvroPayload(HoodieKey key, String commitTime) throws IOException {
-    GenericRecord rec = generateGenericRecord(key.getRecordKey(), "rider-" + commitTime, "driver-" + commitTime, 0.0);
+    GenericRecord rec = generateGenericRecord(key.getRecordKey(), "rider-" + commitTime, "driver-" + commitTime, 0);
     return new HoodieAvroPayload(Option.of(rec));
   }
 
   public static GenericRecord generateGenericRecord(String rowKey, String riderName, String driverName,
-      double timestamp) {
+      long timestamp) {
```

Comment:
Is the reason for this change to use the timestamp to generate partition for which a double value is not conducive ?

Reply:
Yes, I tried to change the type of `timestamp` field to `double`. However, it will cause the test case: `TestHoodieTestSuiteJob#testComplexDag` failed.

Reply:
Yes. I have tried to change the type of `timestamp` field of `source.avsc` file to `doubule`. However,  it will cause `TestHoodieTestSuiteJob#testComplexDag` to be failed.

Reply:
Okay, we can fix that, shouldn't be difficult